src/charts/stacked-bar.js

```diff
@@ -0,0 +1,1010 @@
+            top: 70,
+            right: 30,
+            bottom: 60,
+            left: 70
+        },
+        width = 960,
+        height = 500,
+
+        xScale,
+        xAxis,
+        yScale,
+        yAxis,
+
+        aspectRatio = null,
+
+        verticalTicks = 5,
+        yTickTextYOffset = -8,
+        yTickTextXOffset = -20,
+
+        numOfVerticalTicks = 5,
+        numOfHorizontalTicks = 5,
+        percentageAxisToMaxRatio = 1,
+
+        colorSchema = colorHelper.colorSchemas.britechartsColorSchema,
+
+        colorScale,
+        categoryColorMap,
+
+        layers,
+        layersInitial,
+
+        verticalMarkerContainer,
+
+        ease = d3Ease.easeQuadInOut,
+        horizontal = false,
+
+        svg,
+        chartWidth, chartHeight,
+        data,
+        stacks,
+        dataInitial,
+        dataZeroed,
+        transformedData,
+
+        tooltipThreshold = 480,
+
+        xAxisPadding = {
+            top: 0,
+            left: 0,
+            bottom: 0,
+            right: 0
+        },
+        maxBarNumber = 8,
+
+        animationDelayStep = 20,
+        animationDelays = d3Array.range(animationDelayStep, maxBarNumber* animationDelayStep, animationDelayStep),
+        animationDuration = 1000,
+        
+        baseLine,
+        verticalGridLines,
+        horizontalGridLines,
+        grid = null,
+
+        nameLabel = 'name',
+        valueLabel = 'value',
+        stackLabel = 'stack',
+        nameLabelFormat,
+        valueLabelFormat = NUMBER_FORMAT,
+
+            // getters
+            getName = (data) =>  data[nameLabel],
+            getValue = (data) => data[valueLabel],
+            getStack = (data) => data[stackLabel],
+            isAnimated = false,
+
+            // events
+            dispatcher = d3Dispatch.dispatch('customMouseOver', 'customMouseOut', 'customMouseMove');
+
+       /**
+         * This function creates the graph using the selection and data provided
+         * @param {D3Selection} _selection A d3 selection that represents
+         * the container(s) where the chart(s) will be rendered
+         * @param {areaChartData} _data The data to attach and generate the chart
+         */
+         function exports(_selection) {
+            _selection.each(function(_data){
+                chartWidth = width - margin.left - margin.right;
+                chartHeight = height - margin.top - margin.bottom;
+                prepareData(_data);
+                buildScales();
+                buildLayers();
+                buildSVG(this);
+                drawGridLines();
+                buildAxis();
+                drawAxis();
+                drawStackedBar();
+                if (shouldShowTooltip()){   
+                    addMouseEvents();
+                }
+            });
+        }
+        /**
+         * Prepare data for create chart.
+         * @private
+         */
+         function prepareData(_data) {
+            data = cleanData(_data);
```

Comment:
Could we move back this one? We are trying to make it a default method on all charts.
Thanks!